File: 2515_Shortest_Distance_to_Target_String_in_a_Circular_Array/solution.c
int closestTarget(char** w, int ws, char* target, int si) {
    int a = ws, n = ws;
    for (int i = 0; i < n; ++i) if (strcmp(w[i], target) == 0) { int d = abs(i - si); a = fmin(a, fmin(d, n - d)); }
    return a < n ? a : -1;
}

File: 2515_Shortest_Distance_to_Target_String_in_a_Circular_Array/solution.py
class Solution:
    def closestTarget(self, ws: List[str], target: str, si: int) -> int:
        a = n = len(ws)
        for i, w in enumerate(ws):
            if w == target: a = min(a, abs(i - si), n - abs(i - si))
        return a if a < n else -1
